### PR TITLE
Starting the FAQ page

### PR DESCRIPTION
**Pull request by bradygaster**
Initial creation of this page with only a few sections in it, to get the conversation started. Sent review requests to team, and look forward to the feedback and/or contributions of other questions we're going to need in here. 

Goal for this page is to make give the community guidance prior to them submitting an issue; hopefully this will result in some of what-would-have-been issues being mitigated.

### CODE REVIEW

File: FAQ.md [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+
+**What is SignalR?**
+
+SignalR is a series of libraries for both the server and client that make it incredibly simple to add real-time web functionality to your apps. You can learn more about SignalR in the [SignalR documentation](https://docs.microsoft.com/en-us/aspnet/core/signalr/introduction). 
+
+**How does SignalR enable "real-time HTTP?"**
+
+There are two sets of open-source libraries that make this possible: 
+
+1. The server-side package, which is used by .NET developers to build server-side SignalR Hubs. This package is distributed via [NuGet](https://nuget.org) in the [Microsoft.AspNetCore.SignalR](https://www.nuget.org/packages/Microsoft.AspNetCore.SignalR) package. This server-side package is not required if your applications make use of the [SignalR Azure Service](https://azure.microsoft.com/en-us/services/signalr-service/).
```
Comment (anurse) [inline FAQ.md]: Users shouldn't see the server-side package at all unless they are using .NET Framework. I'd just say it's distributed as part of ASP.NET Core (in the `Microsoft.AspNetCore.App` package)

Comment (bradygaster) [reply FAQ.md]: @anurse made changes to your much-appreciated feedback. Let me know if you have more to add, looking forward to it.

**Comment by Eilon** [comment]
And another option would be maybe to add this to the docs repo instead?

File: FAQ.md [hunk 2]
```diff
@@ -0,0 +1,49 @@
+SignalR Frequently Asked Questions
+=======
+
+This document will condense some of the more popular questions and themes that have come up in the issues we've received on SignalR. Please check the FAQ before you submit a new issue, as the issue may have already been addressed below. 
+
+---
+
+**What is SignalR?**
+
+SignalR is a series of libraries for both the server and client that make it incredibly simple to add real-time web functionality to your apps. You can learn more about SignalR in the [SignalR documentation](https://docs.microsoft.com/en-us/aspnet/core/signalr/introduction).
```
Comment (muratg) [inline FAQ.md]: Isn't it more than "web" now? E.g. Java client on Android, and serverless -- no web.

File: FAQ.md [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+            }, 5000);
+        });
+};
+
+connection.onclose(function () {
+    start();
+});
+
+start();
+```
```
Comment (muratg) [inline FAQ.md]: Should we consider moving this to TypeScript/ES6 style? (try, await, catch.)

File: FAQ.md [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+
+No. SignalR clients need to be manually reconnected. Below you will see some example JavaScript code that demonstrates one method of performing a reconnection. 
+
+```javascript
+const start = () => {
+    connection.start()
+        .then(() => {
+            console.log('connected');
+        })
+        .catch(err => {
```
Comment (muratg) [inline FAQ.md]: Nit: `catch(err) { ... `

File: FAQ.md [hunk 5]
```diff
@@ -0,0 +1,49 @@
...
+                start();
+            }, 5000);
+        });
+};
+
+connection.onclose(function () {
+    start();
+});
+
+start();
```
Comment (muratg) [inline FAQ.md]: I'd write this as:
```javascript
const start = () => {
  try {
    await connection.start()
    console.log('connected')
  } catch (err) {
    console.log(err)
    setTimeout(() => start(), 5000)
  }
}
```

File: FAQ.md (hunk 2 above)
Comment (bradygaster) [reply FAQ.md]: couldn't agree more. update OTW will have this removed.

File: FAQ.md (hunk 4 above)
Comment (bradygaster) [reply FAQ.md]: i approve of your nit and condone your OCD.

File: FAQ.md (hunk 5 above)
Comment (bradygaster) [reply FAQ.md]: Thanks. Way better.

File: FAQ.md (hunk 3 above)
Comment (bradygaster) [reply FAQ.md]: i think this way is fine, but i'm open to it if that's *the new normal*.

**Comment by bradygaster** [comment]
@Eilon good feedback, and the FAQ will probably have numerous links to the docs that go deeper on various areas, but the idea here is that we've seen a lot of issues that are repeating. So we wanted to offer up a FAQ that represents issues we've observed the community having frequently. They could check here before submitting an issue we've already addressed.

**Comment by bradygaster** [comment]
@Eilon that said - I **do** prefer the idea of this being in a wiki page instead of as a markdown file in the root.

File: FAQ.md [hunk 6]
```diff
@@ -0,0 +1,44 @@
...
+1. Developers who are using .NET Core on the server side to build custom real-time endpoints use [SignalR Hubs](https://docs.microsoft.com/en-us/aspnet/core/signalr/hubs), which are available as a part of .NET Core (via the [ASP.NET Core NuGet](https://www.nuget.org/packages/Microsoft.AspNetCore.App)). 
+
+1. Developers who simply need the real-time endpoint and who want to use custom client-side code to handle all events coming from the server should use of the [SignalR Azure Service](https://azure.microsoft.com/en-us/services/signalr-service/). 
+
+**Does SignalR Reconnect Automatically?**
+
+No. SignalR clients need to be manually reconnected. Below you will see some example JavaScript code that demonstrates one method of performing a reconnection. 
+
+```javascript
+const start = () => {
```
Comment (anurse) [inline FAQ.md]:
```suggestion
const start = async () => {
```

File: FAQ.md [hunk 7]
```diff
@@ -0,0 +1,44 @@
...
+
+1. Developers who are using .NET Core on the server side to build custom real-time endpoints use [SignalR Hubs](https://docs.microsoft.com/en-us/aspnet/core/signalr/hubs), which are available as a part of .NET Core (via the [ASP.NET Core NuGet](https://www.nuget.org/packages/Microsoft.AspNetCore.App)). 
+
+1. Developers who simply need the real-time endpoint and who want to use custom client-side code to handle all events coming from the server should use of the [SignalR Azure Service](https://azure.microsoft.com/en-us/services/signalr-service/). 
+
+**Does SignalR Reconnect Automatically?**
+
+No. SignalR clients need to be manually reconnected. Below you will see some example JavaScript code that demonstrates one method of performing a reconnection. 
+
+```javascript
```
Comment (anurse) [inline FAQ.md]:
```suggestion
```typescript
```

File: FAQ.md (hunk 3 above)
Comment (anurse) [reply FAQ.md]: Yeah, show the new hotness, but tell people that this is using TypeScript/ES6 syntax, so they need to transpile it if they aren't already.

**Comment by bradygaster** [comment]
Created the [FAQ page](https://github.com/aspnet/SignalR/wiki/Frequently-Asked-Questions) in the wiki. Will close this PR. Makes sense to keep it in the wiki. Thanks all!

**Pull request closed without merging**